nixos/modules/flyingcircus/packages/fcsensuplugins/fc/sensuplugins/journal.py

```diff
@@ -0,0 +1,115 @@
+""" Journal check.
+
+Takes a set of regular expression, tries to match them with the output of
+journalctl and if positive prints them.
+"""
+
+import argparse
+import logging
+import nagiosplugin
+import re
+import subprocess
+import yaml
+
+_log = logging.getLogger('nagiosplugin')
+
+
+class Journal(nagiosplugin.Resource):
+
+    def __init__(self, params, src):
+        self.params = params
+        self.journalctl = self.params.journalctl
+        self.src = src
+
+        self.warnings = []
+        self.criticals = []
+
+    def find_hits(self):
+
+        critical_hits = []
+        warning_hits = []
+        log = subprocess.check_output(self.journalctl, shell=True).decode().splitlines()
+        with open(self.src) as fobj:
+                    patterns = yaml.load(fobj)
+
+        # create a list of match candidates by applying the rules
+        warning_hits = [
+            entry
+            for rule in patterns['warningpatterns']
+            for entry in log
+            if re.search(rule, entry)
+        ]
+        # reducing them by exceptions of the rule
+        if warning_hits: warning_hits = [
```

Comment:
Please format Python source according to PEP8 (and install a check plugin into your editor!)